rest-api/test/unit_test/api_test/participant_summary_api_test.py

```diff
@@ -1093,6 +1094,130 @@ def testWithdrawThenPair(self):
+                                                CONSENT_PERMISSION_YES_CODE, time=TIME_2)
+
+    ps_1 = self.send_get('Participant/%s/Summary' % participant_id_1)
+    # the enrollmentStatusMemberTime should still be TIME_1
+    self.assertEquals(TIME_1.isoformat(), ps_1.get('enrollmentStatusMemberTime'))
+
+  def test_dv_consent_after_ehr_consent(self):
+    questionnaire_id_1 = self.create_questionnaire('dv_ehr_share_consent_questionnaire.json')
+    questionnaire_id_2 = self.create_questionnaire('ehr_consent_questionnaire.json')
+    participant_1 = self.send_post('Participant', {})
+    participant_id_1 = participant_1['participantId']
+
+    with FakeClock(TIME_6):
+      self.send_consent(participant_id_1)
+
+    # submit ehr consent only
+    self._submit_consent_questionnaire_response(participant_id_1, questionnaire_id_2,
+                                                CONSENT_PERMISSION_YES_CODE, time=TIME_1)
+
+    ps_1 = self.send_get('Participant/%s/Summary' % participant_id_1)
+    # the enrollmentStatusMemberTime should still be TIME_1
+    self.assertEquals(TIME_1.isoformat(), ps_1.get('enrollmentStatusMemberTime'))
+
+    # submit dv consent after ehr consent at TIME_2
+    self._submit_dvehr_consent_questionnaire_response(participant_id_1, questionnaire_id_1,
+                                                      DVEHRSHARING_CONSENT_CODE_YES, time=TIME_2)
+
+    ps_1 = self.send_get('Participant/%s/Summary' % participant_id_1)
+    # the enrollmentStatusMemberTime should still be TIME_1
+    self.assertEquals(TIME_1.isoformat(), ps_1.get('enrollmentStatusMemberTime'))
+
+  def test_dv_consent_withdraw_ehr_consent(self):
+    questionnaire_id_1 = self.create_questionnaire('dv_ehr_share_consent_questionnaire.json')
+    questionnaire_id_2 = self.create_questionnaire('ehr_consent_questionnaire.json')
+    participant_1 = self.send_post('Participant', {})
+    participant_id_1 = participant_1['participantId']
+
+    with FakeClock(TIME_6):
+      self.send_consent(participant_id_1)
+
+    # submit dv consent only, the enrollmentStatusMemberTime should be TIME_1
+    self._submit_dvehr_consent_questionnaire_response(participant_id_1, questionnaire_id_1,
+                                                      DVEHRSHARING_CONSENT_CODE_YES, time=TIME_1)
+
+    ps_1 = self.send_get('Participant/%s/Summary' % participant_id_1)
+    self.assertEquals(TIME_1.isoformat(), ps_1.get('enrollmentStatusMemberTime'))
+
+    # withdraw ehr consent after dv consent at TIME_2
+    self._submit_consent_questionnaire_response(participant_id_1, questionnaire_id_2,
+                                                CONSENT_PERMISSION_NO_CODE, time=TIME_2)
+
+    ps_1 = self.send_get('Participant/%s/Summary' % participant_id_1)
+    # ehr consent overwrite dv consent, enrollmentStatusMemberTime should be None
+    self.assertIsNone(ps_1.get('enrollmentStatusMemberTime'))
```

Comment:
add assertion for enrollmentStatus and consentForElectronicHealthRecords as well ?

Reply:
done